### PR TITLE
feat(low-code cdk): add dynamic stream config to check connection

### PR DESCRIPTION
**Pull request by lazebnyi**
Fixed: https://github.com/airbytehq/airbyte-internal-issues/issues/12083 

This PR modifies the check connections functionality and adds dynamic stream check configurations, allowing users to specify check settings in a single component for different stream types.

### CODE REVIEW

File: airbyte_cdk/sources/declarative/checks/check_stream.py [hunk 1]
```diff
@@ -23,34 +33,126 @@ class CheckStream(ConnectionChecker):
...
         self, source: AbstractSource, logger: logging.Logger, config: Mapping[str, Any]
     ) -> Tuple[bool, Any]:
-        streams = source.streams(config=config)
+        """Checks the connection to the source and its streams."""
+        try:
+            streams = source.streams(config=config)
+            if not streams:
+                return False, f"No streams to connect to from source {source}"
+        except Exception as error:
+            return self._log_error(logger, "connecting to streams", error)
```
Comment (maxi297) [inline airbyte_cdk/sources/declarative/checks/check_stream.py:58]: nit: It feels more aligned with our language to say "discovering streams" instead of "connecting to streams"

**Review by maxi297 — APPROVED** [review]
I have one question but this looks good to me

File: airbyte_cdk/sources/declarative/checks/check_stream.py (hunk 1 above)
Comment (lazebnyi) [reply airbyte_cdk/sources/declarative/checks/check_stream.py]: updated

File: airbyte_cdk/sources/declarative/declarative_component_schema.yaml [hunk 2]
```diff
@@ -330,6 +329,28 @@ definitions:
...
+        enum: [ DynamicStreamCheckConfig ]
+      dynamic_stream_name:
+        title: Dynamic Stream Name
+        description: The dynamic stream name.
+        type: string
+      stream_count:
+        title: Stream Count
+        description: Numbers of the streams to try reading from when running a check operation.
+        type: integer
+        default: 0
```
[the remark this answers is not in the document]
Comment (lazebnyi) [reply airbyte_cdk/sources/declarative/declarative_component_schema.yaml:353]: > Why have a default value of 0? It seems weird to validate 0 streams.  

If the value is 0, we skip running `read` for the static stream generated by this dynamic stream, assuming that only successfully created streams are enough to pass the check for this dynamic stream.  

> Also, in which case would we want to check more than 1 stream? It feels dangerous, as users might only have one stream for a specific dynamic stream configuration.  

We need to ensure that all or at least more than one of the streams described by the configuration values are valid. That’s why we require more than one. (Perhaps we could reference a configuration field to fetch the size automatically and validate the streams. This could be a next step in extending connection checks for dynamic streams.)  

Inside the `_check_generated_streams_availability` method, we handle cases where the stream count exceeds the length of the generated streams by taking the minimum of the two values.

[the remark this answers is not in the document]
Comment (maxi297) [reply airbyte_cdk/sources/declarative/declarative_component_schema.yaml:353]: Ok cool, thanks for the precision. Can we make this part of the description or some doc somewhere? 

Also, I don't see a test where we're taking the minimum of the two values. Should we add that?

**Review by maxi297 — APPROVED** [review]
Looks good to me! Thanks for the clarifications
